src/observers/link_prefetch_observer.js

```diff
@@ -100,14 +100,11 @@ export class LinkPrefetchObserver {
 
     request.headers["Sec-Purpose"] = "prefetch"
 
-    if (link.dataset.turboFrame && link.dataset.turboFrame !== "_top") {
-      request.headers["Turbo-Frame"] = link.dataset.turboFrame
-    } else if (link.dataset.turboFrame !== "_top") {
-      const turboFrame = link.closest("turbo-frame")
+    const turboFrame = link.closest("turbo-frame")
+    const turboFrameTarget = link.dataset.turboFrame || turboFrame?.getAttribute("target") || turboFrame?.id
```

Comment:
Could we rely on `getAttribute("data-turbo-frame")` instead of the `dataset` property?

Most other places in the application, we use the `getAttribute` method to fetch attributes. If we use `.dataset` here, grepping for `data-turbo-frame` won't include this line.